cigaradvisor/blocks/promotion-banner/promotion-banner.js

```diff
@@ -0,0 +1,99 @@
+// noinspection JSUnusedGlobalSymbols
+
+import { createOptimizedPicture, readBlockConfig } from '../../scripts/aem.js';
+import { isInternal } from '../../scripts/scripts.js';
+
+const sheetURL = '/cigaradvisor/promotions.json?sheet=promotions';
+
+let promotions;
+
+async function fetchPromotions() {
+  try {
+    // fetch the JSON data
+    const response = await fetch(sheetURL);
+
+    if (!response.ok) {
+      promotions = [];
+      return;
+    }
+
+    const { data } = await response.json();
+    promotions = data;
```

Comment:
should we return variable instead of using global variable to bring more predictability ? @Kronnox @bstopp

Reply:
Can there be more than one promotion block on any given page?

Reply:
yes..like home page

Reply:
Not sure what you mean by `more predictability`, but the global variable is used here to create a global site cache for the promotions. As the block is used in multiple places the data only needs to be fetched once this way. At least that was the idea behind it.

Alternatively we could even go as far as storing it in the local storage and have an invalidation schema, but I think that's overkill tbh.

Reply:
ok..we can merge..

Reply:
@bstopp any objections?

Reply:
is it possible that the same promotion can show up twice on one page ?